// ngx-photo-editor/esm2020/lib/ngx-photo-editor.module.mjs
import { NgModule } from '@angular/core';
import { NgxPhotoEditorComponent } from './ngx-photo-editor.component';
import { CommonModule } from "@angular/common";
import * as i0 from "@angular/core";
export class NgxPhotoEditorModule {
}
NgxPhotoEditorModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.2", ngImport: i0, type: NgxPhotoEditorModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxPhotoEditorModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.0.2", ngImport: i0, type: NgxPhotoEditorModule, declarations: [NgxPhotoEditorComponent], imports: [CommonModule], exports: [NgxPhotoEditorComponent] });
NgxPhotoEditorModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.0.2", ngImport: i0, type: NgxPhotoEditorModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.2", ngImport: i0, type: NgxPhotoEditorModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NgxPhotoEditorComponent
                    ],
                    imports: [
                        CommonModule,
                    ],
                    exports: [
                        NgxPhotoEditorComponent
                    ]
                }]
        }] });
