// ngx-photo-editor/esm2020/lib/ngx-photo-editor.service.mjs
import { Injectable } from '@angular/core';
import { Subject } from "rxjs";
import { NgxPhotoEditorComponent } from "./ngx-photo-editor.component";
import * as i0 from "@angular/core";
export class NgxPhotoEditorService {
    constructor(componentFactoryResolver, appRef, injector) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.appRef = appRef;
        this.injector = injector;
    }
    open(source, options) {
        const componentFactory = this.componentFactoryResolver.resolveComponentFactory(NgxPhotoEditorComponent);
        const componentRef = componentFactory.create(this.injector);
        this.appRef.attachView(componentRef.hostView);
        const domElem = componentRef.hostView.rootNodes[0];
        document.body.appendChild(domElem);
        this.ngxPESubscriber = new Subject();
        this.ngxPEComponentRef = componentRef;
        this.ngxPEComponentRef.instance.closeEvent.subscribe(() => this.close());
        this.ngxPEComponentRef.instance.errorEvent.subscribe((data) => this.errorHandler(data));
        this.ngxPEComponentRef.instance.imageCroppedEvent.subscribe((data) => this.export(data));
        if (options) {
            Object.keys(options).map(value => {
                // @ts-ignore
                this.ngxPEComponentRef.instance[value] = options[value];
            });
        }
        this.ngxPEComponentRef.instance.source = source;
        return this.ngxPESubscriber.asObservable();
    }
    errorHandler(data) {
        this.ngxPESubscriber.error(data);
        this.close();
    }
    close() {
        this.ngxPESubscriber.complete();
        this.ngxPEComponentRef.destroy();
    }
    export(data) {
        this.ngxPESubscriber.next(data);
        this.close();
    }
}
NgxPhotoEditorService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.2", ngImport: i0, type: NgxPhotoEditorService, deps: [{ token: i0.ComponentFactoryResolver }, { token: i0.ApplicationRef }, { token: i0.Injector }], target: i0.ɵɵFactoryTarget.Injectable });
NgxPhotoEditorService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.0.2", ngImport: i0, type: NgxPhotoEditorService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.2", ngImport: i0, type: NgxPhotoEditorService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i0.ComponentFactoryResolver }, { type: i0.ApplicationRef }, { type: i0.Injector }]; } });
